tests/test_signals.py

```diff
@@ -46,6 +46,30 @@ def record(sender, template, context):
     finally:
         flask.template_rendered.disconnect(record, app)
 
+def test_before_render_template():
+    app = flask.Flask(__name__)
+
+    @app.route('/')
+    def index():
+        return flask.render_template('simple_template.html', whiskey=42)
+
+    recorded = []
+
+    def record(sender, template, context):
+        context['whiskey'] = 43
+        recorded.append((template, context))
+
+    flask.before_render_template.connect(record, app)
```

Comment:
You can use the [context manager](https://pythonhosted.org/blinker/#blinker.base.Signal.connected_to) to connect the signal temporarily